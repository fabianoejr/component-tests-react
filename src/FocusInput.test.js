import React from "react";
import FocusInput from "../src/FocusInput";
import { render, fireEvent, cleanup } from "@testing-library/react";

afterEach(cleanup);

test("FocusInput matches snapshot", () => {
  const { container } = render(<FocusInput />);
  expect(container.firstChild).toMatchSnapshot();
});

test("clicking on button trigger focus on input", () => {
  const { getByPlaceholderText, getByText } = render(<FocusInput />);
  fireEvent.click(getByText("Clique aqui para focar"));
  expect(document.activeElement).toBe(getByPlaceholderText("Foco aqui!!!"));
});
