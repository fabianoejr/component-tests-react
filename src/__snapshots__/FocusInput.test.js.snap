// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusInput matches snapshot 1`] = `
<div>
  <input
    aria-label="focus-input"
    placeholder="Foco aqui!!!"
    type="text"
  />
  <button>
    Clique aqui para focar
  </button>
</div>
`;
